core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -377,6 +382,48 @@ private Status parseToStatus(@Nonnull final IMAPResponse[] r) throws ParsingExce
             return status;
         }
 
+        /**
+         * Parses the LIST-STATUS command responses to a list of ListStatus objects. See RFC 5819 for details. For each selectable mailbox matching
+         * the list pattern and selection options, the server MUST return an untagged LIST response followed by an untagged STATUS response containing
+         * the information requested in the STATUS return option. If an attempted STATUS for a listed mailbox fails because the mailbox can't be
+         * selected , the STATUS response MUST NOT be returned and the LIST response MUST include the \NoSelect attribute.
+         *
+         * @param r the list of responses from Status command, the input responses array should contain the tagged/final one
+         * @return Status object constructed based on the r array
+         * @throws ParsingException when encountering parsing exception
+         * @throws ImapAsyncClientException when input value is not valid
+         */
+        @Nonnull
+        private ListStatusList parseToListStatusList(@Nonnull final IMAPResponse[] r) throws ParsingException, ImapAsyncClientException {
+            if (r.length < 1) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            final Response taggedResponse = r[r.length - 1];
+            if (!taggedResponse.isOK()) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            // command successful reaching here
+            final List<ListStatus> v = new ArrayList<ListStatus>();
+
+            ListInfo linfo = null;
+            for (int i = 0; i < r.length; i++) {
+                final IMAPResponse ir = r[i];
+
+                if (ir.keyEquals("LIST")) {
```

Comment:
Seems like you are missing the case when last response is a list response.

Reply:
thank u! Fixed!